File: Leetcode/go/611.go
package main

import "sort"

func triangleNumber(nums []int) int {
	if len(nums) < 3 {
		return 0
	}

	sort.Ints(nums)

	res := 0
	for i := len(nums) - 1; i >= 2; i-- {
		left, right := 0, i-1

		for left < right {
			if nums[left]+nums[right] > nums[i] {
				res += right - left
				right--
			} else {
				left++
			}
		}
	}

	return res
}

File: Leetcode/611.cs
namespace Leetcode
{
    internal class _611
    {
        public int TriangleNumber(int[] nums)
        {
            var res = 0;
            if (nums.Length < 3)
            {
                return res;
            }

            Array.Sort(nums);

            for (int i = nums.Length - 1; i >= 2; i--)
            {
                int left = 0,
                    right = i - 1;
                while (left < right)
                {
                    if (nums[left] + nums[right] > nums[i])
                    {
                        res += right - left;
                        right--;
                    }
                    else
                    {
                        left++;
                    }
                }
            }

            return res;
        }
    }
}
